### PR TITLE
=act #15040 ScatterGatherFirstCompleted reply when no routees

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #15040

File: akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala [hunk 1]
```diff
@@ -86,4 +89,15 @@ class ScatterGatherFirstCompletedSpec extends AkkaSpec with DefaultTimeout with
 
   }
 
+  "Scatter-gather pool" must {
+
+    "without routees should reply immediately" in {
+      val probe = TestProbe()
+      val router = system.actorOf(ScatterGatherFirstCompletedPool(nrOfInstances = 0, within = 5.seconds).props(Props.empty))
```
Comment (ktoso) [inline akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala]: I thought that we may consider `0` illegal here, but the case Heiko spoke of can well be triggered by all the initially in-the-pool routees to die, leaving the pool having 0 routees... then the optimisation makes sense, right?

Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala]: Yes, i just thought this was an easy way to test it

Comment (ktoso) [inline akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala]: Okey, sounds good. Making the 0 throw right away during construction probably does not buy much here in terms of safety for users, they'll notice anyway.

**Comment by ktoso** [comment]
LGTM

File: akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala (hunk 1 above)
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala]: There might be corner cases with not allowing 0, e.g. could work with 0 + resizer.

Comment (ktoso) [inline akka-actor-tests/src/test/scala/akka/routing/ScatterGatherFirstCompletedSpec.scala]: I see :) The PR looks good in any case, feel free to merge when you get the time
